labelbox/data/serialization/coco/instance_dataset.py

```diff
@@ -0,0 +1,183 @@
+# https://cocodataset.org/#format-data
+
+from labelbox.data.serialization.coco.categories import Categories, hash_category_name
+from labelbox.data.serialization.coco.annotation import COCOObjectAnnotation, RLE, get_annotation_lookup, rle_decoding
+from labelbox.data.serialization.coco.image import CocoImage, get_image, get_image_id
+from typing import Any, Dict, List
+from pydantic import BaseModel
+from ...annotation_types import ImageData, MaskData, Mask, ObjectAnnotation, Label, Polygon, Point, Rectangle
+import numpy as np
+from PIL import Image
+from tqdm import tqdm
+import os
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from ...annotation_types.collection import LabelCollection
+
+
+def mask_to_coco_object_annotation(annotation: ObjectAnnotation, annot_idx,
+                                   image_id, category_id):
+    # This is going to fill any holes into the multipolygon
+    # If you need to support holes use the panoptic data format
+    shapely = annotation.value.shapely.simplify(1).buffer(0)
+    xmin, ymin, xmax, ymax = shapely.bounds
+    # Iterate over polygon once or multiple polygon for each item
+    area = shapely.area
+    if shapely.type == 'Polygon':
+        shapely = [shapely]
+
+    return COCOObjectAnnotation(
+        id=annot_idx,
+        image_id=image_id,
+        category_id=category_id,
+        segmentation=[
+            np.array(s.exterior.coords).ravel().tolist() for s in shapely
+        ],
+        area=area,
+        bbox=[xmin, ymin, xmax - xmin, ymax - ymin],
+        iscrowd=0)
+
+
+def vector_to_coco_object_annotation(annotation: ObjectAnnotation, annot_idx,
+                                     image_id: int, category_id):
+    shapely = annotation.value.shapely
+    xmin, ymin, xmax, ymax = shapely.bounds
+    segmentation = []
+    if isinstance(annotation.value, Polygon):
+        for point in annotation.value.points:
+            segmentation.extend([point.x, point.y])
+    else:
+        box = annotation.value
+        segmentation.extend([
+            box.start.x, box.start.y, box.end.x, box.start.y, box.end.x,
+            box.end.y, box.start.x, box.end.y
+        ])
+
+    return COCOObjectAnnotation(id=annot_idx,
+                                image_id=image_id,
+                                category_id=category_id,
+                                segmentation=[segmentation],
+                                area=shapely.area,
+                                bbox=[xmin, ymin, xmax - xmin, ymax - ymin],
+                                iscrowd=0)
+
+
+def rle_to_common(class_annotations, class_name):
+    mask = rle_decoding(class_annotations.segmentation.counts,
+                        *class_annotations.segmentation.size[::-1])
+    return ObjectAnnotation(name=class_name,
```

Comment:
Not I was going to leave a comment "I don't see `name` in `ObjectAnnotation` but then realized that `BaseAnnotation` inherits from `FeatureSchema`. Imho it's quite confusing, it should be a composition rather than inheritance.